packages/cogify/src/cogify/cli/cli.topo.ts

```diff
@@ -0,0 +1,207 @@
+import { Bounds, Epsg, Nztm2000Tms, TileMatrixSets } from '@basemaps/geo';
+import { fsa, LogType } from '@basemaps/shared';
+import { CliInfo } from '@basemaps/shared/build/cli/info.js';
+import { boolean, command, flag, option, string } from 'cmd-ts';
+import pLimit from 'p-limit';
+
+import { isArgo } from '../../argo.js';
+import { UrlFolder } from '../../cogify/parsers.js';
+import { getLogger, logArguments } from '../../log.js';
+import { TopoStacItem } from '../stac.js';
+import { groupTiffsByDirectory, mapEpsgToSlug } from '../topo/mapper.js';
+import { createStacCollection, createStacItems, writeStacFiles } from '../topo/stac.creation.js';
+import { brokenTiffs } from '../topo/types.js';
+
+const Q = pLimit(10);
+
+/**
+ * List all the tiffs in a directory for topographic maps and create cogs for each.
+ *
+ * @param source: Location of the source files
+ * @example s3://linz-topographic-upload/topographic/TopoReleaseArchive/NZTopo50_GeoTif_Gridless/
+ *
+ * @param target: Location of the target path
+ */
+export const TopoStacCreationCommand = command({
+  name: 'cogify-topo-stac',
+  version: CliInfo.version,
+  description: 'List input topographic files, create StacItems, and generate tiles for grouping.',
+  args: {
+    ...logArguments,
+    title: option({
+      type: string,
+      long: 'title',
+      description: 'Imported imagery title',
+    }),
+    source: option({
+      type: UrlFolder,
+      long: 'source',
+      description: 'Location of the source files',
+    }),
+    target: option({
+      type: UrlFolder,
+      long: 'target',
+      description: 'Target location for the output files',
+    }),
+    scale: option({
+      type: string,
+      long: 'scale',
+      description: 'topo25, topo50, or topo250',
+    }),
+    resolution: option({
+      type: string,
+      long: 'resolution',
+      description: 'e.g. gridless_600dpi',
+    }),
+    latestOnly: flag({
+      type: boolean,
+      defaultValue: () => false,
+      long: 'latest-only',
+      description: 'Only process the latest version of each map sheet',
+      defaultValueIsSerializable: true,
+    }),
+    forceOutput: flag({
+      type: boolean,
+      defaultValue: () => false,
+      long: 'force-output',
+      defaultValueIsSerializable: true,
+    }),
+  },
+  async handler(args) {
+    const logger = getLogger(this, args);
+    const startTime = performance.now();
+    logger.info('ListJobs:Start');
+
+    const { epsgDirectoryPaths, stacItemPaths } = await loadTiffsToCreateStacs(
+      args.latestOnly,
+      args.source,
+      args.target,
+      args.title,
+      args.scale,
+      args.resolution,
+      args.forceOutput,
+      logger,
+    );
+
+    if (epsgDirectoryPaths.length === 0 || stacItemPaths.length === 0) throw new Error('No Stac items created');
+
+    // write stac items into an JSON array
+    if (args.forceOutput || isArgo()) {
+      const targetURL = isArgo() ? fsa.toUrl('/tmp/topo-stac-creation/') : args.target;
```

Comment:
generally its best to not upper case variables `targetURL -> targetUrl` 

As you get into weird screaming words when you join two or more together `HTMLURL`